docs/src/adr/005adr-json.md

```diff
@@ -95,21 +162,27 @@ The `value` field depends on the kind of `TlaValue`:
 1. For `TlaBool`: a JSON Boolean
 1. For `TlaInt(bigIntValue)`: 
   1. If `bigIntValue.isValidInt`: a JSON number
-  2. Otherwise: `{ bigInt: bigIntValue.toString() }`
+  2. Otherwise: `{ "bigInt: bigIntValue.toString() }`
 1. For `TlaDecimal(decValue)`: a JSON string `decValue.toString`
 
+The reason for the non-uniform treatment of integers, is that Apalache encodes its TLA+ integers as `BigInt`, which means that it permits values, for which `.isValidInt` does not hold. 
+
+While it might seem more natural to encode the entire `TlaValue` as a JSON primitive, without the added object layer, we would have a much tougher time deserializing.
+We would need a) a sensible encoding of `BigInt` values, which are not valid integers, and b) a way to distinguish both variants of `BigInt`, as well as decimals, when deserializing (since JSON is not typed).
+We could encode all values as strings, but they would be similarly indistinguishable when deserializing. Importantly, the `type` field of the `ValEx` expression is not guaranteed to contain a hint, as it could be `Untyped`
+
 Take `jsonOf1` to be the serialization of `ValEx( TlaInt( 1 ) )( typeTag = Typed( IntT1() ) )` shown above. The serialization of `1 + 1` is then equal to
 ```
 {
-  kind: "OperEx",
-  type: "Int",
-  oper: "(+)",
-  args: [jsonOf1, jsonOf1]
+  "kind": "OperEx",
+  "type": "Int",
+  "oper": "(+)",
```

Comment:
```suggestion
  "oper": "+",
```

`(+)` is a different operator in TLA+.

Reply:
Perhaps, but that is the value of `TlaArithOper.plus.name`

Reply:
Since it is an ADR and we have #726, let's change it to `PLUS` right away.